src/jit/flowgraph.cpp

```diff
@@ -17069,6 +17098,9 @@ BasicBlock* Compiler::fgFindInsertPoint(unsigned    regionIndex,
             regionIndex, dspBool(putInTryRegion), startBlk->bbNum, (endBlk == nullptr) ? 0 : endBlk->bbNum,
             (nearBlk == nullptr) ? 0 : nearBlk->bbNum, (jumpBlk == nullptr) ? 0 : jumpBlk->bbNum, dspBool(runRarely));
 
+    const bool insertingIntoFilter = !putInTryRegion && (startBlk == ehGetDsc(regionIndex - 1)->ebdFilter) &&
+                                     (endBlk == ehGetDsc(regionIndex - 1)->ebdHndBeg);
+
```

Comment:
I assume you hoisted this because it looked loop-insensitive. However, you've introduced a bug where we're accessing ebdFilter even if the EH region does not represent a filter, which is illegal (since it is a union; we really should assert if we could). Below, we can do this because we know at that point `blk == startBlk` and `blk` is the first block of a filter, and the function requires the caller to pass exactly a filter region as startBlk/endBlk in this case. You could also use `!putInTryRegion && ehGetDsc(regionIndex - 1)->HasFilter() && ...` (but now the number of duplicated `ehGetDsc()` is getting ridiculous)